### PR TITLE
PIA-1003: Sort and filter servers list by favorite and recommended

### PR DESCRIPTION
**Pull request by kp-laura-sempere**
This PR includes search, sort and filter functionatities:
- Search regions by name or country
- Recommended servers (based on latency)
- Store last searched servers

### CODE REVIEW

File: PIA VPN-tvOS/RegionsSelection/Presentation/RegionsListViewModel.swift [hunk 1]
```diff
@@ -11,41 +11,247 @@ import PIALibrary
...
+        onServerSelectedRouterAction.execute()
+    }
+
+    func viewDidAppear() {
+        if filter == .previouslySearched {
+            refreshRegionsList()
+        }
+    }
+    
+    private func refreshRegionsList() {
```
Comment (kp-said-rehouni) [inline PIA VPN-tvOS/RegionsSelection/Presentation/RegionsListViewModel.swift:63]: It seems that this ViewModel is probably doing many things, maybe we can break it down a bit, what do you think?

**Review by kp-said-rehouni — COMMENTED** [review]
Great job @kp-laura-sempere! Just a couple of comments that we can discuss tomorrow in a session.

File: PIA VPN-tvOS/RegionsSelection/Presentation/RegionsListViewModel.swift (hunk 1 above)
Comment (kp-laura-sempere) [reply PIA VPN-tvOS/RegionsSelection/Presentation/RegionsListViewModel.swift:63]: Thanks @kp-said-rehouni , 
Yes,  the thing is that we need to perform a lot of things from the regions list: add and remove to favorites, sorting by name and by recommended and search. 
I created various UseCases to handle getting the servers list and also adding and removing from favorites. And this VM makes use of those use cases to do that. But in terms of sorting and filtering, not sure how we can split it out of this VM.
Do you have any suggestions?

**Review by kp-said-rehouni — APPROVED** [review]
LGTM! Great job @kp-laura-sempere!

**Pull request merged**